lib/applications/resolvers.ts

```diff
@@ -547,3 +549,95 @@ export const createReplacementApplication: Resolver = async (_, args, { prisma }
     ok: true,
   };
 };
+
+export const generateApplicantsReport: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  // NOTETOSELF: Consider making columns an object instead to avoid array look up inefficiency + clean up code
+
+  const applications: GenerateApplicantsReportApplications[] = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: new Date(startDate),
+        // TODO; Figure out why lte isn't working
+        // lte: new Date(endDate),
+      },
+    },
+    select: {
+      // TODO?: Replace with function/object in constants
+      rcdUserId: columns.includes(ApplicationsReportColumn.UserId),
+      firstName: columns.includes(ApplicationsReportColumn.ApplicantName),
+      middleName: columns.includes(ApplicationsReportColumn.ApplicantName),
+      lastName: columns.includes(ApplicationsReportColumn.ApplicantName),
+      dateOfBirth: columns.includes(ApplicationsReportColumn.ApplicantDob),
+      createdAt: columns.includes(ApplicationsReportColumn.ApplicationDate),
+      paymentMethod: columns.includes(ApplicationsReportColumn.PaymentMethod),
+      processingFee:
+        columns.includes(ApplicationsReportColumn.FeeAmount) ||
+        columns.includes(ApplicationsReportColumn.TotalAmount),
+      donationAmount:
+        columns.includes(ApplicationsReportColumn.DonationAmount) ||
+        columns.includes(ApplicationsReportColumn.TotalAmount),
+      permits: {
+        select: {
+          rcdPermitId: true,
+        },
+      },
+    },
+  });
+
+  // Adds totalAmount and applicantName properties to applications object
+  applications.map(application => {
+    application.applicantName =
+      application.firstName +
+      (application.middleName
+        ? ` ${application.middleName} ${application.lastName}`
+        : ` ${application.lastName}`);
+    application.totalAmount = (application.processingFee || 0) + (application?.donationAmount || 0);
+    application.rcdPermitId = application.permits?.rcdPermitId;
+    return application;
+  });
+
+  const csvHeaders = [];
+
+  if (columns.includes(ApplicationsReportColumn.UserId)) {
```

Comment:
https://github.com/uwblueprint/richmond-centre-for-disability/pull/109/files#r757935443